plugins/inputs/win_services/win_services.go

```diff
@@ -151,17 +165,30 @@ func (m *WinServices) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
-// listServices returns a list of services to gather.
-func listServices(scmgr WinServiceManager, userServices []string) ([]string, error) {
-	if len(userServices) != 0 {
-		return userServices, nil
+func (m *WinServices) createServicesFilter() error {
+	filter, err := filter.NewIncludeExcludeFilter(m.ServiceNames, nil)
+	if err != nil {
+		return err
 	}
+	m.servicesFilter = filter
+	return nil
+}
```

Comment:
can probably collapse this all into one line, might not need a function for it.

Reply:
Updated like suggested, thanks!